src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -431,28 +432,116 @@ private static bool OverloadResolutionResultIsValid<TMember>(ArrayBuilder<Member
         }
 
 #nullable enable
-        private void RemoveCallingConventionMismatches<TMember>(ArrayBuilder<MemberResolutionResult<TMember>> results, Cci.CallingConvention expectedConvention) where TMember : Symbol
+        private void RemoveCallingConventionMismatches<TMember>(ArrayBuilder<MemberResolutionResult<TMember>> results, in CallingConventionInfo expectedConvention) where TMember : Symbol
         {
             if (typeof(TMember) != typeof(MethodSymbol))
             {
                 return;
             }
 
+            Debug.Assert(expectedConvention.CallingConventionTypes is not null);
+
             for (int i = 0; i < results.Count; i++)
             {
                 var result = results[i];
                 var member = (MethodSymbol)(Symbol)result.Member;
                 if (result.Result.IsValid)
                 {
-                    if (!member.CallingConvention.IsCallingConvention(expectedConvention)
-                        || member.CallingConvention.HasUnknownCallingConventionAttributeBits())
+                    if (member.CallingConvention.HasUnknownCallingConventionAttributeBits())
+                    {
+                        results[i] = makeWrongCallingConvention(result);
+                        continue;
+                    }
+
+                    var unmanagedCallersOnlyData = member.UnmanagedCallersOnlyAttributeData;
+                    if (ReferenceEquals(unmanagedCallersOnlyData, UnmanagedCallersOnlyAttributeData.AttributePresentDataNotBound)
+                        || ReferenceEquals(unmanagedCallersOnlyData, UnmanagedCallersOnlyAttributeData.Uninitialized))
+                    {
+                        // If we're at a location where the unmanaged data has actually not yet been bound, it cannot be
+                        // valid code anyway, as the only place that can occur is when we're actually doing an address of
```

Comment:
This assumption doesn't feel right. I think it is quite possible to attempt to access a member for which we haven't bound attributes.